pkg/ebpf/c/tracee.bpf.c

```diff
@@ -8036,9 +8055,16 @@ CGROUP_SKB_HANDLE_FUNCTION(proto_icmpv6)
 CGROUP_SKB_HANDLE_FUNCTION(proto_tcp_dns)
 {
     // submit DNS base event if needed (full packet)
-    if (should_submit_net_event(neteventctx, SUB_NET_PACKET_DNS)) {
+    if (should_submit_net_event(neteventctx, SUB_NET_PACKET_DNS))
         cgroup_skb_submit_event(ctx, neteventctx, NET_PACKET_DNS, FULL);
-        cgroup_skb_capture(); // capture tcp dns packets
+
+    if (should_submit_net_event(neteventctx, SUB_NET_PACKET_IP) ||
+        should_submit_net_event(neteventctx, SUB_NET_PACKET_TCP) ||
+        should_submit_net_event(neteventctx, SUB_NET_PACKET_DNS)) {
+
+        // full dns header for capture (payload doesn't make much sense)
+        neteventctx->md.header_size = ctx->len;
+        cgroup_skb_capture(); // capture dns-tcp, tcp or ip packets
```

Comment:
nit: order

```suggestion
        cgroup_skb_capture(); // capture ip, tcp or dns-tcp packets
```

Reply:
This is in reverse order because that is how it should be going backwards (in the logic of capturing).